capstone_project/src/main/java/com/google/sps/data/Post.java

```diff
@@ -0,0 +1,136 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import com.google.appengine.api.datastore.Text;
+import java.util.Map;
+
+public class Post {
+  private String text;
+  private String fileType;
+  private String title;
+  private String schoolName;
+  private String fileBlobKey;
+  private Map<String, Object> reactions;
+  private long upvotes;
+  private long postId;
+
+  /**
+   * Builder class for posts. Creates a Post object with properties: text, fileType, title,
+   * schoolName, fileBlobKey, reactions, upvotes, and post id.
+   */
+  public static class Builder {
+    private String text;
+    private String fileType;
+    private String title;
+    private String schoolName;
+    private String fileBlobKey;
+    private Map<String, Object> reactions;
+    private long upvotes;
+    private long postId;
+
+    /** Sets the text property. Converts the given Text instance into a string. */
+    public Builder setText(Text text) {
+      this.text = text.getValue();
+      return this;
+    }
+
+    public Builder setFileType(String fileType) {
```

Comment:
please add comments to all of these. I know it's kinda annoying but let's just dot our i-s and cross our t-s

Reply:
Done